evm-adapters/src/sputnik/cheatcodes/memory_stackstate_owned.rs

```diff
@@ -150,6 +153,47 @@ impl<'config, B: Backend> MemoryStackStateOwned<'config, B> {
     }
 }
 
+impl<'config, B: Backend> FuzzState for MemoryStackStateOwned<'config, B> {
+    fn flatten_state(&self) -> HashSet<[u8; 32]> {
+        let mut flattened: HashSet<[u8; 32]> = HashSet::new();
+        let (applies, logs) = self.substate.clone().deconstruct(&self.backend);
+        for apply in applies {
```

Comment:
Wonder if we should add a test case that shows this working? Or some alternative strategy instead of flattening the state across many types into H256s, given that this misses type information? e.g. it means that you may populate the dictionary with a Address value and consume it as a Bytes somewhere, where it might not be required?

Reply:
added test case